arch/arm/core/aarch32/Kconfig

```diff
@@ -221,6 +221,7 @@ choice
 
 config FP_HARDABI
 	bool "Floating point Hard ABI"
+	depends on !BUILD_WITH_TFM
```

Comment:
This is quite a big limitation, could we (perhaps in help text) inform the users why hard fp cannot be used ?

Would also be great if we somewhere (docs ? ) can place a link to any open TF-M issues or tasks around this limitation,

Reply:
> Not sure I follow, hard abi is causing build failures when using TF-M, so we would not be dropping support for anything cause the support has never been there when enabling TF-M.

I meant that if might be possible to identify what parts of TFM exactly cause the build failures, when using hard ABI in Zephyr, isolate them, and offer hard ABI support with TFM, if these parts are not used. That's my idea at least - not sure about how practical this is.

Reply:
Got it, that would indeed be nice if it were possible, but looking at the top-level CMakeLists.txt file in TF-M, that's gonna be hard:
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/d07d72c64e4b3d8594829eea302b88e51709861f/trusted-firmware-m/CMakeLists.txt#L29-L31
```
if(TFM_SYSTEM_FP)
    message(FATAL_ERROR "Hardware FPU is currently not supported in TF-M")
endif()
```

Reply:
@tejlmand a comment was added to explain the limitation.